include/drivers/gpio.h

```diff
@@ -451,6 +457,21 @@ struct gpio_callback {
  *
  * For internal use only, skip these in public documentation.
  */
+
+/* Used by driver api function pin_interrupt_configure, these are defined
+ * in terms of the public int flags so we can just mask and pass them
+ * through to the driver api
+ */
+enum gpio_int_trigger_mode {
+	GPIO_INT_MODE_DISABLED = 0,
+	GPIO_INT_MODE_LEVEL_LOW = GPIO_INT_LOW_0,
+	GPIO_INT_MODE_LEVEL_HIGH = GPIO_INT_HIGH_1,
+	GPIO_INT_MODE_EDGE_FALLING = GPIO_INT_EDGE | GPIO_INT_LOW_0,
+	GPIO_INT_MODE_EDGE_RISING = GPIO_INT_EDGE | GPIO_INT_HIGH_1,
+	GPIO_INT_MODE_EDGE_BOTH = GPIO_INT_EDGE | GPIO_INT_LOW_0 |
+				  GPIO_INT_HIGH_1,
+};
```

Comment:
This can be implemented simpler as
```
	GPIO_INT_MODE_DISABLED = GPIO_INT_DISABLE,
	GPIO_INT_MODE_LEVEL_LOW = GPIO_INT_LEVEL_LOW,
	GPIO_INT_MODE_LEVEL_HIGH = GPIO_INT_LEVEL_HIGH,
	GPIO_INT_MODE_EDGE_FALLING = GPIO_INT_EDGE_FALLING,
	GPIO_INT_MODE_EDGE_RISING = GPIO_INT_EDGE_RISING,
	GPIO_INT_MODE_EDGE_BOTH = GPIO_INT_EDGE_BOTH,
```